src/Evolution/Systems/NewtonianEuler/Sources.hpp

```diff
@@ -0,0 +1,37 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace NewtonianEuler {
+
+/*!
+ * \brief Compute the source terms for the NewtonianEuler evolution
+ * using a problem-specific source of type `SourceTermType`
+ */
+template <typename SourceTermType>
+struct ComputeSources {
+  using return_tags = typename SourceTermType::return_tags;
+
+  using argument_tags = tmpl::push_front<typename SourceTermType::argument_tags,
+                                         Tags::SourceTerm<SourceTermType>>;
+
+  static constexpr size_t volume_dim = SourceTermType::volume_dim;
+
+  template <class... Args>
+  static void apply(
+      gsl::not_null<tnsr::I<DataVector, volume_dim>*> source_momentum_density,
+      gsl::not_null<Scalar<DataVector>*> source_energy_density,
+      const db::item_type<Tags::SourceTerm<SourceTermType>>& source,
+      Args&... args) noexcept {
```

Comment:
Is `Args&` intended here, or should it be `const Args&` or `Args&&`?

Reply:
I think here I just chose the one that wasn't giving compilation errors. Which one should be the preferred here?

Reply:
`const Args&` should be correct if it compiles.  If that doesn't work I'll take a closer look at exactly how this is being used.

Reply:
`const Args&` worked. Not sure if/why I tried that before :P